cilium-cli/connectivity/tests/encryption_v2.go

```diff
@@ -110,52 +110,138 @@ func (s *podToPodEncryptionV2) Name() string {
 
 // resolveEgressDevice resolves the egress device used in the provided host
 // network namespace used to send traffic to dst.
-func (s *podToPodEncryptionV2) resolveEgressDevice(ctx context.Context, srcHostNS *check.Pod, dst *check.Pod) (string, error) {
+func (s *podToPodEncryptionV2) resolveEgressDevice(ctx context.Context, srcHostNS *check.Pod, src, dst *check.Pod) (string, error) {
+	// defaultIPRouteGet issues `ip route get` for destination in provided host network
+	// namespace and extract device.
+	//
+	// example json output:
+	// [{"dst":"192.168.109.96","gateway":"192.168.128.1","dev":"ens5","prefsrc":"192.168.159.49","flags":[],"uid":0,"cache":[]}]
+	defaultIPRouteGet := func(dstIP string) (string, error) {
+		out, err := srcHostNS.K8sClient.ExecInPod(ctx,
+			srcHostNS.Pod.Namespace,
+			srcHostNS.Pod.Name,
+			"",
+			[]string{"ip", "-j", "route", "get", dstIP})
+
+		if err != nil {
+			return "", fmt.Errorf("Failed to resolve egress device for: %w", err)
+		}
+
+		routes := []struct {
+			Dev string `json:"dev,omitempty"`
+		}{}
+
+		err = json.Unmarshal(out.Bytes(), &routes)
+		if err != nil {
+			return "", fmt.Errorf("Failed to parse ip route to json: %w", err)
+		}
+
+		for _, route := range routes {
+			if route.Dev != "" {
+				return route.Dev, nil
+			}
+		}
+
+		return "", fmt.Errorf("Failed to find egress device")
+	}
+
+	// tableIPRouteGet issues `ip route show table X` in provided host network
+	// namespace and extract device.
+	//
+	// example json output:
+	// [{"dst":"default","gateway":"192.168.128.1","dev":"ens6","flags":[]},{"dst":"192.168.128.1","dev":"ens6","scope":"link","flags":[]}]
+	tableIPRouteGet := func(table string) (string, error) {
```

Comment:
This function is specifically looking for the 'device for the default route' but comments and function name don't really mention this.

Reply:
You're absolutely right, didn't update comments in the code after changing the logic :sweat_smile: 
Just adjusted the comments, thanks!